transport/wseb/src/main/java/org/kaazing/gateway/transport/wseb/WsebSession.java

```diff
@@ -393,12 +421,16 @@ public void attachReader(final HttpSession newReader) {
             if (ALIGN_UPSTREAM) {
                 final Thread ioThread = getIoThread();
                 final Executor ioExecutor = getIoExecutor();
+                // Prevent messageReceived from being fired from setIoAlignment and racing ahead of attachReader0
+                newReader.suspendRead();
```

Comment:
As we discussed today, I think this is ok. There are guards you put in idle timeout (if session is closed or current thread is unset).Closing shouldn't be an issue because nothing happens after that.